src/icrar/leap-accelerate/ms/MeasurementSet.cc

```diff
@@ -32,30 +34,102 @@ namespace icrar
     , m_filepath(filepath)
     , m_readAutocorrelations(readAutocorrelations)
     {
-        m_stations = overrideNStations.is_initialized() ? overrideNStations.get() : m_measurementSet->antenna().nrow();
+        // Check and use unique antennas 
+        casacore::Vector<casacore::Int> a1 = m_msmc->antenna1().getColumn();
+        casacore::Vector<casacore::Int> a2 = m_msmc->antenna1().getColumn();
+        auto a1s = std::set<int32_t>(a1.cbegin(), a1.cend());
+        auto a2s = std::set<int32_t>(a2.cbegin(), a2.cend());
+        std::set_union(a1.cbegin(), a1.cend(), a2.cbegin(), a2.cend(), std::inserter(m_antennas, m_antennas.begin()));
```

Comment:
If `m_antennas` was a local variable instead of a member (see other related comment) this could go inside the `else if` block.

Also see my other comment about code duplication.

Reply:
I've removed a number of the redundant calculations now